src/field.cpp

```diff
@@ -2496,19 +2496,10 @@ field_id field_entry::setFieldType(const field_id new_field_id){
 
 }
 
-int field_entry::setFieldDensity(const int new_density){
-
-    if(new_density > 3) {
-        density = 3;
-    } else if (new_density < 1) {
-        density = 1;
-        is_alive = false;
-    } else {
-        density = new_density;
-    }
-
-    return density;
-
+int field_entry::setFieldDensity(const int new_density)
+{
+    is_alive = new_density > 1;
+    return std::max( std::min( new_density, MAX_FIELD_DENSITY ), 1 );
```

Comment:
`this->density` is not changed by the function anymore. The function name is a lie.